### ISSUE TITLE
[v4] Documentation

### ISSUE THREAD

**Issue by negezor** [negezor/vk-io#255]
## TL;DR
At the moment, the library is refactoring with major changes. And there is a problem with non-written documentation and a description of existing features.

## General stuff
- [ ] Choose a platform for documentation, select (`docpress`/`docsify`/`vuepress`)
- [X] API Reference generation
#### @vk-io/authorization
- [ ] Authorization example (Direct)
- [ ] Authorization example (Implicit Flow User)
- [ ] Authorization example (Implicit Flow Group)
- [ ] Using proxies

#### @vk-io/scenes
- [ ] Using session and state
- [ ] Create your own scene (Advanced)
- [ ] General example with a global middleware handler (Advanced)
- - Step scenes
- - [ ] Step scene example
- - [ ] Moving from one handler to another
- - [ ] Example `enterHandler` and `leaveHandler`

#### @vk-io/session
- [ ] Basic example
- [ ] Custom session key to the storage
- [ ] Implementing your own storage (Advanced)

#### @vk-io/streaming
- [ ] Basic example
- [ ] An example of working with rules (CRUD)

#### vk-io
- [ ] Basic example
- [ ] Using proxies
- **API**
- [ ] Basic example
- [ ] Description of different operating modes
- [ ] Example with `call` and note why not use that
- [ ] Example with `execute` and `procedure`
- [ ] Call using `APIRequest` (Advanced)
- **Collect**
- [ ] Basic example
- [ ] Interaction with options
- [ ] Using readable stream
- **Snippets**
- [ ] Example with `resolveResource`
- **Updates**
- [ ] Basic example
- [ ] Examples of working with `use`
- [ ] Examples of working with `on`
- [ ] Handle events directly (Advanced)
- [ ] Description of work with middleware (Advanced)
- - **Polling**
- - [ ] The difference between `User Long Poll` and` Bots Long Poll`
- - **Webhook**
- - [ ] Web server launch
- - [ ] Integration with [express](https://github.com/expressjs/express)/[koa](https://github.com/koajs/koa)
- - **Contexts**
- - [ ] Description of the basic principles of working with the context
- **Upload**
- [ ] Basic example
- [ ] Description of the basic principles
- **Attachments**
- [ ] Description of the basic principles of working with the attachments
- **Keyboard**
- [ ] Basic example
- [ ] Example with a builder (KeyboardBuilder)
- [ ] Alternative keyboard example (basic class Keyboard)

## Notice
This question combines the previous issue on the documentation, kind of this is my todo list. As well as the ability to help with documentation using Pull Request.
